x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -392,4 +397,34 @@ public void testTranslateMathFunction_HavingClause_Painless() {
         assertThat(aggFilter.scriptTemplate().params().toString(), startsWith("[{a=MAX(int){a->"));
         assertThat(aggFilter.scriptTemplate().params().toString(), endsWith(", {v=10}]"));
     }
-}
+
+    public void testGroupByDateHistogram() {
+        LogicalPlan p = plan("SELECT MAX(int) FROM test GROUP BY HISTOGRAM(int, 1000)");
```

Comment:
Do we have a situation where the histogram is being grouped by very small intervals? Like 1 second for a date interval.

Reply:
I tried a test with 1 second and it worked :). Our data might not be representative but I think it's a side effect of composite aggs which paginates through data instead of creating all the buckets up-front.